geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala

```diff
@@ -24,15 +27,31 @@ class HBaseDataStoreFactory extends DataStoreFactorySpi {
 
   import HBaseDataStoreFactory.Params._
 
+  // TODO: can we have multiple connections in a single JVM?
+  private class Key(val connection: Connection, val config: HBaseDataStoreConfig) {
+    override def hashCode(): Int = config.hashCode()
+    override def equals(obj: scala.Any): Boolean = config.equals(obj)
+  }
+
+  private val dsCache =
+    CacheBuilder.newBuilder().build(
+      new CacheLoader[Key, HBaseDataStore] {
+        override def load(k: Key): HBaseDataStore = {
+          new HBaseDataStore(k.connection, k.config)
```

Comment:
dunno if we should cache these... then wouldn't `dispose()` affect other data stores? seems like we could manage the datastore lifecycle better in spark instead